ssh/util/src/main/java/org/wildfly/security/ssh/util/SshUtil.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.wildfly.security.ssh.util;
+
+import org.wildfly.common.iteration.ByteIterator;
+import org.wildfly.common.iteration.CodePointIterator;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import org.apache.sshd.common.config.keys.FilePasswordProvider;
+import org.apache.sshd.common.config.keys.loader.openssh.OpenSSHKeyPairResourceParser;
+import org.wildfly.security.pem.Pem;
+import org.wildfly.security.pem.PemEntry;
+
+import static org.wildfly.security.ssh.util._pivate.ElytronMessages.log;
+
+/**
+ * Class providing utilities for parsing of keys in OpenSSH format
+ */
+public class SshUtil {
+
+    public static final String OPENSSH_PRIVATE_KEY_FORMAT = "OPENSSH PRIVATE KEY";
+
+
+    /**
+     * Iterate over the contents of a key file in OpenSSH format, returning each entry in sequence.
+     *
+     * @param pemContent the code point iterator over the content (must not be {@code null})
+     * @param passphraseProvider provides the passphrase used to decrypt the private key(may be {@code null})
+     * @return the iterator (not {@code null})
+     * @throws IllegalArgumentException if there is a problem with the data or the key
+     *
+     */
+    public static Iterator<PemEntry<?>> parsePemOpenSSHContent(CodePointIterator pemContent, FilePasswordProvider passphraseProvider) throws IllegalArgumentException {
```

Comment:
I think we're missing a check to make sure `pemContent` is not null.

Reply:
@fjuma Added. thank you